### PR TITLE
AP-305 AP-339 Dashboard and Vpn panel buttons ux fixes

### PR DESCRIPTION
**Pull request by ravjit-cliqz**

### CODE REVIEW

File: app/src/cliqz/java/com/cliqz/browser/controlcenter/ControlCenterHelper.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ public void hideControlCenter() {
     }
 
     @Override
-    public void toggleControlCenter() {
+    public void toggleControlCenter(Bus bus) {
```
Comment (spacifici) [inline app/src/cliqz/java/com/cliqz/browser/controlcenter/ControlCenterHelper.java]: Why?

File: app/src/cliqz/java/com/cliqz/browser/vpn/VpnPanel.java [hunk 2]
```diff
@@ -19,4 +20,12 @@ public static VpnPanel create(View view) {
 
     public void show(FragmentManager fragmentManager, String tag) {
     }
+
+    public boolean isVisible() {
+        return false;
```
Comment (spacifici) [inline app/src/cliqz/java/com/cliqz/browser/vpn/VpnPanel.java]: ?

File: app/src/cliqz/java/com/cliqz/browser/vpn/VpnPanel.java [hunk 3]
```diff
@@ -19,4 +20,12 @@ public static VpnPanel create(View view) {
...
+    }
+
+    public Dialog getDialog() {
+        return null;
```
Comment (spacifici) [inline app/src/cliqz/java/com/cliqz/browser/vpn/VpnPanel.java]: ?

File: app/src/lumen/java/com/cliqz/browser/controlcenter/ControlCenterHelper.kt [hunk 4]
```diff
@@ -19,9 +21,17 @@ class ControlCenterHelper(private val fragmentManager: FragmentManager) : Contro
...
 
-    override fun toggleControlCenter() {
+    override fun toggleControlCenter(bus: Bus) {
+        if (controlCenterDialog != null && controlCenterDialog!!.isVisible) {
+            controlCenterDialog?.dialog?.dismiss()
+            return
+        }
         controlCenterDialog = ControlCenterDialog.create(mSource)
         controlCenterDialog?.show(fragmentManager, Constants.CONTROL_CENTER)
+        var handler = Handler()
```
Comment (spacifici) [inline app/src/lumen/java/com/cliqz/browser/controlcenter/ControlCenterHelper.kt]: We may like to use an injected handler.

File: app/src/lumen/java/com/cliqz/browser/vpn/VpnPanel.java [hunk 5]
```diff
@@ -136,6 +136,7 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
         bus.register(this);
     }
 
+
```
Comment (spacifici) [inline app/src/lumen/java/com/cliqz/browser/vpn/VpnPanel.java]: extra line

File: app/src/main/java/com/cliqz/browser/controlcenter/ControlCenterActions.java [hunk 6]
```diff
@@ -2,11 +2,13 @@
...
 import android.view.View;
 
+import com.cliqz.nove.Bus;
+
 public interface ControlCenterActions {
 
     void hideControlCenter();
 
-    void toggleControlCenter();
+    void toggleControlCenter(Bus bus);
```
Comment (spacifici) [inline app/src/main/java/com/cliqz/browser/controlcenter/ControlCenterActions.java]: Why do you pass the bus via a method?

File: app/src/main/java/com/cliqz/browser/widget/SearchBar.java [hunk 7]
```diff
@@ -376,6 +376,8 @@ public void onFocusChange(View v, boolean hasFocus) {
 
             if (hasFocus) {
                 post(SearchBar.this::showKeyBoard);
+                bus.post(new Messages.DismissVpnPanel());
```
Comment (spacifici) [inline app/src/main/java/com/cliqz/browser/widget/SearchBar.java]: Why two messages?

File: app/src/main/java/com/cliqz/browser/controlcenter/ControlCenterActions.java [hunk 8]
```diff
@@ -2,6 +2,8 @@
 
 import android.view.View;
 
+import com.cliqz.nove.Bus;
```
Comment (spacifici) [inline app/src/main/java/com/cliqz/browser/controlcenter/ControlCenterActions.java]: Extra import

File: app/src/main/java/com/cliqz/browser/widget/SearchBar.java (hunk 7 above)
Comment (ravjit-cliqz) [reply app/src/main/java/com/cliqz/browser/widget/SearchBar.java]: We have to dismiss vpn panel or dashboard when we tap on url bar. Either of them could be open.

File: app/src/main/java/com/cliqz/browser/controlcenter/ControlCenterActions.java (hunk 8 above)
Comment (ravjit-cliqz) [reply app/src/main/java/com/cliqz/browser/controlcenter/ControlCenterActions.java]: Updated

**Pull request merged**